File: Argus/Angular-Frontend/src/app/model/person.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import {environment} from '../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class PersonService {
  private baseUrl = `${environment.apiUrl}/springboot/api/people`;

  constructor(private http: HttpClient) { }

  getPersonById(id: number): Observable<any> {
    return this.http.get(`${this.baseUrl}/${id}`);
  }

  addPerson(person: any): Observable<any> {
    return this.http.post(`${this.baseUrl}`, person);
  }

  updatePerson(id: number, value: any): Observable<any> {
    return this.http.put(`${this.baseUrl}/${id}`, value);
  }

  deletePerson(id: number): Observable<any> {
    return this.http.delete(`${this.baseUrl}/${id}`, { responseType: 'text' });
  }

  getPersonList(): Observable<any> {
    return this.http.get(`${this.baseUrl}`);
  }
}

File: Spring-Backend/src/main/java/com/springboot/SpringBackend/service/PersonService.java
package com.springboot.SpringBackend.service;

import com.springboot.SpringBackend.model.Person;

import java.util.List;
import java.util.Optional;

public interface PersonService {
    List<Person> getAllPeople();

    Optional<Person> getPersonById(Long id);

    Person createPerson(Person x);

    Person updatePerson(Person x);

    void deletePerson(Person x);

    void deleteAllPeople();
}
